internal/api/schemas/v1.yaml

```diff
@@ -11,17 +11,168 @@ info:
     name: Mozilla Public License 2.0
     url: https://www.mozilla.org/en-US/MPL/2.0/
 paths:
+  "/namespaces/{namespace}/gateways":
+    get:
+      description: 'Returns all configured gateways in the given namespace.'
+      operationId: listNamespacedGateways
+      parameters:
+      - name: namespace
+        in: path
+        description: Namespace of gateways to fetch
+        required: true
+        schema:
+          type: string
+      responses:
+        '200':
+          description: gateway response
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/GatewayPage"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Error"
   "/gateways":
     get:
       description: 'Returns all configured gateways.'
       operationId: listGateways
+      parameters:
+      - name: namespaces
+        in: query
+        description: The namespaces of gateways to list
+        schema:
+          type: string
+      responses:
+        '200':
+          description: gateway response
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/GatewayPage"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Error"
+    post:
+      description: 'Creates a new gateway.'
+      operationId: createGateway
+      requestBody:
+        description: Gateway to create
+        required: true
+        content:
+          application/json:
+            schema:
+              "$ref": "#/components/schemas/Gateway"
+      responses:
+        '201':
+          description: gateway response
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Gateway"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Error"
+  "/namespaces/{namespace}/gateways/{name}":
```

Comment:
I could see us needing a `PUT` on this path, but I'm thinking you might be holding off on that until we discuss `PUT` vs. `PATCH` in the future?

Reply:
Yeah, that and I'm also leaning towards making the `Create` endpoint more of an "upsert" endpoint in the long run (since you can identify gateways/routes without their path parameters due to them returning names/namespaces in their bodies) with PUT semantics more than anything -- which would eventually mean we could extend this to include PATCH, but wouldn't have to initially since we already have the ability to update.

Reply:
Fine punting on this for now